### ISSUE TITLE
Value cannot be null. (Parameter 'o')

### ISSUE THREAD

**Issue by Trapulo** [dotnet/efcore#26559]
Hi,
I have the error "System.ArgumentNullException: Value cannot be null. (Parameter 'o')" running update-dabase, and I cannot find any way to understand or debug the problem.

```
System.ArgumentNullException: Value cannot be null. (Parameter 'o')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationCommandListBuilder.Append(String o)
   at Microsoft.EntityFrameworkCore.Migrations.SqlServerMigrationsSqlGenerator.Generate(AlterDatabaseOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.<>c.<.cctor>b__83_5(MigrationsSqlGenerator g, MigrationOperation o, IModel m, MigrationCommandListBuilder b)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.SqlServerMigrationsSqlGenerator.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(IReadOnlyList`1 operations, IModel model, MigrationsSqlGenerationOptions options)
   at Microsoft.EntityFrameworkCore.Migrations.SqlServerMigrationsSqlGenerator.Generate(IReadOnlyList`1 operations, IModel model, MigrationsSqlGenerationOptions options)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.GenerateUpSql(Migration migration, MigrationsSqlGenerationOptions options)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.<>c__DisplayClass16_2.<GetMigrationCommandLists>b__2()
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.UpdateDatabase(String targetMigration, String connectionString, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.UpdateDatabaseImpl(String targetMigration, String connectionString, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.UpdateDatabase.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)

-- verbose doesn't show anything useful

EF Core version: 5.0.9
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 5.0
Operating system: Windows 10
IDE: Visual Studio 2019 16.11.5

any idea?
Thanks

**Comment by roji** [comment]
@Trapulo can you please submit a code sample which reproduces the above? It's difficult to understand what's going on from an exception stack trace and no additional context.

**Comment by Trapulo** [comment]
I know. But I have an up migration of about 3300 rows and a the designer.cs class is about 4300 rows (with sensible table names, etc). 
This is one of reason I cannot find a why to debug or understand. All "my" code is executed without error, then the runtime error come from EF as black box.

Any idea I can extract something useful?

**Comment by ajcvickers** [comment]
@Trapulo Could be a duplicate of #24328. Please try using the latest [daily build](https://github.com/dotnet/efcore/blob/main/docs/DailyBuilds.md) and report back if it still throws.

**Comment by Trapulo** [comment]
So I tried to:

- install VS 2022
- upgrade all solution to .NET 6
- run update-database

The problem was not gone, but the new ED instead of that usefulness message started to generate clear errors. For example "System.InvalidOperationException: To change the IDENTITY property of a column, the column needs to be dropped and recreated   at Microsoft.EntityFrameworkCore.Migrations.SqlServerMigrationsSqlGenerator.Generate(AlterColumnOperation operation, IModel model, MigrationCommandListBuilder builder)"

and some other error dropping non-existent indexes in my migration.

So I was able to fix them also in .NET 5 and run migration.

thanks
